### PR TITLE
Track downloads in fathom

### LINKED ISSUE #460: Add Fathom Analytics Event Tracking to Downloads
## Problem
We would like to track downloads of Processing for each platform (Linux, macOS (Intel), macOS (Apple Silicon), Raspberry Pi (32-bit), Raspberry Pi (64-bit), Windows) using Fathom Analytics.

## Proposed Solution
We can achieve this by adding Fathom's `trackGoal` method to the event handler for each download button. This method will send an event to Fathom each time a user downloads Processing for a specific platform.

Here are the Fathom event IDs for each platform:

- Download Processing Linux: HHYWFK7G
- Download Processing macOS (Apple Silicon): IWSPGL5F
- Download Processing macOS (Intel): VQUBVEQR
- Download Processing Raspberry Pi (32-bit): ZKSBBVWD
- Download Processing Raspberry Pi (64-bit): TXVODVYO
- Download Processing Windows: CIMDWXJV

## Details
The click handler for the download buttons should be modified to include the Fathom event tracking code. Here's an example of what the handler might look like:

```javascript
const handleDownload = (platform) => {
  switch (platform) {
    case 'Windows':
      fathom.trackGoal('CIMDWXJV', 0);
      break;
    case 'macOS (Intel)':
      fathom.trackGoal('VQUBVEQR', 0);
      break;
    case 'macOS (Apple Silicon)':
      fathom.trackGoal('IWSPGL5F', 0);
      break;
    case 'Linux':
      fathom.trackGoal('HHYWFK7G', 0);
      break;
    case 'Raspberry Pi (32-bit)':
      fathom.trackGoal('ZKSBBVWD', 0);
      break;
    case 'Raspberry Pi (64-bit)':
      fathom.trackGoal('TXVODVYO', 0);
      break;
    default:
      console.error(`Unsupported platform: ${platform}`);
  }
  // code to start the download goes here
};
```

In this example, `handleDownload` should replace the existing click handler for the download buttons. The actual code to start the download, which is not shown in this example, should replace // code to start the download goes here.

This is just an example and may need to be adapted to fit the actual structure and logic of our website.

## References
Fathom documentation on creating and using events: https://usefathom.com/docs/features/events

## Notes
_This issue was partially AI generated and may not be 100% accurate_

### PR DESCRIPTION
**Pull request by runemadsen**
This PR fixes #460 by tracking download events in Fathom Analytics. I implemented this with the `useGoal` that is a part of the `gatsby-plugin-fathom` package. It's a little annoying that we need to run a hook per goal, but the library handles a lot of stuff around dev/prod and SSR so I'd rather rely on that.

Adding @fdoflorenzano so they can take a look. 

CC @SableRaf

### CODE REVIEW

**Comment by SableRaf** [comment]
Hi @runemadsen and thanks for this!

I tried all events and everything seems to be working. _Note: one event doesn't appear on the screenshot below because of pagination._

image

Once @fdoflorenzano gives this PR the thumbs up on the code side, I'll happily merge and release.

**Comment by SableRaf** [comment]
I had a closer look at the code, and it looks good to me! @fdoflorenzano do let me know if you notice anything that needs changing. In the meantime I'll go ahead with the merge. Thanks again for your help on this @runemadsen.

**Review by fdoflorenzano — COMMENTED** [review]
Looks great and works locally for me too! Just left a little question as a comment.

File: src/templates/download.js [hunk 1]
```diff
@@ -31,7 +32,33 @@ const Download = ({ data }) => {
...
+      } else if (asset.os === 'macOS' && asset.bit === 'Intel 64-bit') {
+        trackMacIntel(0);
+      } else if (asset.os === 'macOS' && asset.bit === 'Apple Silicon') {
+        trackMacSilicon(0);
+      } else if (asset.os === 'Linux') {
+        trackLinux(0);
+      } else if (asset.os === 'Raspberry Pi' && asset.bit === '32-bit') {
+        trackPi32(0);
+      } else if (asset.os === 'Raspberry Pi' && asset.bit === '64-bit') {
+        trackPi64(0);
```
[the remark this answers is not in the document]
Comment (runemadsen) [reply src/templates/download.js:56]: Ah, good catch. I just saw them passing `0` in the documentation and assumed it worked like the official API. We can change that.

[the remark this answers is not in the document]
Comment (SableRaf) [reply src/templates/download.js:56]: Thanks @fdoflorenzano! Feel free to make another PR for this tweak.

**Pull request merged**